packages/ui/src/validators/hooks/useValidatorsList.tsx

```diff
@@ -7,82 +7,53 @@ import { Api } from '@/api'
-  'j4Rh1cHtZFAQYGh7Y8RZwoXbkAPtZN46FmuYpKNiR3P2Dc2oz',
-  'j4RjraznxDKae1aGL2L2xzXPSf8qCjFbjuw9sPWkoiy1UqWCa',
-]
-
-const validatorsWithMembership = [
-  {
-    address: 'j4Rh1cHtZFAQYGh7Y8RZwoXbkAPtZN46FmuYpKNiR3P2Dc2oz',
-    membership: {
-      id: '0',
-      handle: 'alice',
-      rootAccount: 'j4Rh1cHtZFAQYGh7Y8RZwoXbkAPtZN46FmuYpKNiR3P2Dc2oz',
-      controllerAccount: 'j4Rh1cHtZFAQYGh7Y8RZwoXbkAPtZN46FmuYpKNiR3P2Dc2oz',
-      boundAccounts: [],
-      inviteCount: 0,
-      roles: [],
-      isVerified: true,
-      isFoundingMember: true,
-      isCouncilMember: true,
-      createdAt: '2023/08/01',
-      entry: {
-        type: 'genesis',
-      },
-      invitees: [],
-      externalResources: [
-        { source: 'TWITTER', value: 'Alice_twitter' },
-        { source: 'DISCORD', value: 'Alice_discord' },
-        { source: 'TELEGRAM', value: 'Alice_telegram' },
-      ],
-    } as MemberWithDetails,
-  },
-  {
-    address: 'j4RjraznxDKae1aGL2L2xzXPSf8qCjFbjuw9sPWkoiy1UqWCa',
-    membership: {
-      id: '1',
-      handle: 'bob',
-      rootAccount: 'j4RjraznxDKae1aGL2L2xzXPSf8qCjFbjuw9sPWkoiy1UqWCa',
-      controllerAccount: 'j4RjraznxDKae1aGL2L2xzXPSf8qCjFbjuw9sPWkoiy1UqWCa',
-      boundAccounts: [],
-      inviteCount: 0,
-      roles: [],
-      isVerified: true,
-      isFoundingMember: false,
-      isCouncilMember: false,
-      createdAt: '2023/08/02',
-      entry: {
-        type: 'genesis',
-      },
-      invitees: [],
-      externalResources: [
-        { source: 'TWITTER', value: 'Bob_twitter' },
-        { source: 'DISCORD', value: 'Bob_discord' },
-      ],
-    } as MemberWithDetails,
-  },
-  { address: 'j4RuqkJ2Xqf3NTVRYBUqgbatKVZ31mbK59fWnq4ZzfZvhbhbN', membership: undefined },
-  { address: 'j4RxTMa1QVucodYPfQGA2JrHxZP944dfJ8qdDDYKU4QbJCWNP', membership: undefined },
-]
-
-const getMember = (address: Address) =>
-  validatorsWithMembership.find((validator) => validator.address === address)?.membership
+interface ValidatorMembership {
+  membership: MemberWithDetails
+  validatorAccount: Address | undefined
+  isVerifiedValidator: boolean | undefined
+}
 
 export const useValidatorsList = () => {
   const { api } = useApi()
   const [search, setSearch] = useState('')
   const [isVerified, setIsVerified] = useState<Verification>(null)
   const [isActive, setIsActive] = useState<State>(null)
   const [visibleValidators, setVisibleValidators] = useState<Validator[]>([])
+  const [validatorsWithMembership, setValidatorsWithMembership] = useState<ValidatorMembership[]>([])
+
+  const allValidatorAddresses = useFirstObservableValue(
+    () =>
+      api?.query.staking.validators
+        .entries()
+        .pipe(map((entries) => entries.map((entry) => entry[0].args[0].toString()))),
+    [api?.isConnected]
+  )
+
+  const variables = {
+    where: { metadata: { validatorAccount_in: allValidatorAddresses } },
+  }
+
+  const { data, loading, error: err } = useGetMembersWithDetailsQuery({ variables })
```

Comment:
The request shouldn't be send unless `allValidatorAddresses` is defined.
```suggestion
  const { data, loading, error: err } = useGetMembersWithDetailsQuery({ variables, skip: !!allValidatorAddresses })
```